html/host.html

```diff
@@ -3,66 +3,100 @@
     <meta charset="utf-8"/>
     <link href="/assets/stylesheets/main.css" rel="stylesheet">
     <link href="/assets/stylesheets/host.css" rel="stylesheet">
+    <script type="module">
+      import { 
+        fullActivityDescription, 
+        shortActivityDescription, 
+        libraryModelList, 
+        cclDiseaseActivityUrl 
+      } from '/assets/data/host.js';
+
+      document.addEventListener("DOMContentLoaded", () => {
+        // Host page content init
+        document.getElementById("activity-text-full").innerHTML = fullActivityDescription;
+        document.getElementById("activity-text-short").innerHTML = shortActivityDescription;
+        document.getElementById("library-model").innerHTML = libraryModelList;
+        document.getElementById("activity-img").src = cclDiseaseActivityUrl;
+
+        // Open & close modal functionality
+        const readMoreBtn = document.getElementById("read-more-btn");
+        const closeModalBtn = document.getElementById("close-modal-btn");
+        const modalContainer = document.getElementById("modal-container");
+
+        readMoreBtn.onclick = () => {
+          modalContainer.style.display = "block";
+        };
+
+        closeModalBtn.onclick = () => {
+          modalContainer.style.display = "none";
+        };
+
+        window.onclick = (event) => {
+          if (event.target == modalContainer) {
+            modalContainer.style.display = "none";
+          }
+        };
+
+        document.addEventListener("keydown", (event) => {
+          if (modalContainer.style.display === "block" && event.key === "Escape") {
+            modalContainer.style.display = "none";
+          }
+        });
+      });
+    </script>
   </head>
   <body>
 
+    <!-- Modal container with full activity description (entire screen)-->
+    <div id="modal-container" class="modal-container">
+
+      <!-- Actual modal here -->
+      <div id="full-description-modal" class="modal">
+        <div class="modal-text">
+          <p class="modal-activity-title">Disease</p>
+          <p id="activity-text-full" class="activity-text-full"></p>
+        </div>
+        <hr class="modal-delineation"></hr>
+        <button id="close-modal-btn" class="close-modal-btn" type="button">Close</button>
```

Comment:
I use `<input type="button">`, not `<button></button>`.

Either way, this is written as `<button type="button">`, which is the worst of both worlds.

Reply:
Oh gosh, not sure how I overlooked this... Noted & done.